### PR TITLE
[15526] Add filenumber/ssn to 5655 form prefill

### PR DESCRIPTION
**Pull request by camerontesterman**
## Description of change
- Adds last four digits of fileNumber and SSN to the form prefill

## Associated Schema Change PR
https://github.com/department-of-veterans-affairs/vets-json-schema/pull/540

## Original issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/15526

## Things to know about this PR
Tested locally

### CODE REVIEW

File: app/models/form_profiles/va5655.rb
Comment (ericboehs) [inline app/models/form_profiles/va5655.rb:17]: What do you think of indenting this line?

File: config/form_profile_mappings/5655.yml
Comment (ericboehs) [inline config/form_profile_mappings/5655.yml:2]: Should this be all caps? or camelCased? This feels weird. 😕

**Review by ericboehs — COMMENTED** [review]
Should we wait until the vets-json-schema change is merged so we can update the Gemfile with the SHA bump?

If so, it'd be nice to mark this as draft and add a todo to the description.

File: config/form_profile_mappings/5655.yml
Comment (camerontesterman) [reply config/form_profile_mappings/5655.yml:2]: It is kind of weird, but if you are referring to the keys being camelCased and values within the array represented as snake_case that is an established pattern within other form profile mappings.

File: app/models/form_profiles/va5655.rb
Comment (camerontesterman) [reply app/models/form_profiles/va5655.rb:17]: Can you send me a snippet of how you think it should be indented?

File: app/models/form_profiles/va5655.rb
Comment (ericboehs) [inline app/models/form_profiles/va5655.rb:18]:
```suggestion
    (
      BGS::PeopleService.new(user).find_person_by_participant_id[:file_nbr].presence ||
        user.ssn.presence
    )&.last(4)
```

File: config/form_profile_mappings/5655.yml
Comment (ericboehs) [reply config/form_profile_mappings/5655.yml:2]: Oh. So it's coming in as `s_s_n`? 😲

File: app/models/form_profiles/va5655.rb
Comment (camerontesterman) [reply app/models/form_profiles/va5655.rb]: I've implemented your suggestion.

File: config/form_profile_mappings/5655.yml
Comment (camerontesterman) [reply config/form_profile_mappings/5655.yml:2]: so when the frontend hits the in_progress_form endpoint (in this case it would be a GET /v0/in_progress_forms/5655) the keys here will be rendered to the frontend in camelCase as they are intended to.

These could be changed, but I'd rather stick as close as possible to what the DMC is expecting once we get to the POST.

The values within the arrays reference Virtus models and associated methods (i.e. FormIdentityInformation class and the ssn_last_four method) that method gets called to populate the `sSN` key which is what the DMC is expecting.

I could represent this as `ssn: [identity_information, ssn_last_four]` but then on the POST either the frontend would need to alter the key to sSN or the backend would prior to POSTing to the DMC.

I would like to get them to change it, but for right now it should stay as it is in my opinion.

**Review by ericboehs — APPROVED** [review]

**Pull request merged**
